### PR TITLE
Minor changes to the default network parameters

### PR DESCRIPTION
**Pull request by SavvasMel**
## Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

Closes https://github.com/ecmwf/WeatherGenerator/issues/294

## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

-   [x] I ran the `uv run train` and (if necessary) `uv run evaluate` on a least one GPU node and it works 
-   [x] If the new feature introduces modifications at the config level, I have made sure to have notified the other software developers through Mattermost and updated the paths in the `$WEATHER_GENERATOR_PRIVATE` directory

### Dependencies

-   [ ] I have ensured that the code is still pip-installable after the changes and runs
-   [ ] I have tested that new dependencies themselves are pip-installable.
-   [ ] I have not introduced new dependencies in the inference portion of the pipeline

### Documentation

-   [ ] My code follows the style guidelines of this project
-   [ ] I have updated the documentation and docstrings to reflect the changes
-   [ ] I have added comments to my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by tjhunter** [comment]
@mlangguth89 this is a larger discussion that is worth to have soon indeed. We are building a sophisticated training pipeline and we should be able to represent it. My suggestion would be to delay this discussion a few days since it has many consequences on how we train and share models.

File: config/default_config.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ pred_mlp_adaln: True
 
 # number of steps offset applied to first target window; if set to zero and forecast_steps=0 then
 # one is training an auto-encoder
-forecast_offset : 1
```
Comment (clessig) [inline config/default_config.yml:41]: This is unrelated to improving the skill--we ask a different question when we change from forecasting to masking. I think the changes in the network parameter configs that you want to implement were with forecasting, right?

**Review by clessig — APPROVED** [review]
Thanks for picking this up. Could you clarify why you want to switch from forecasting to masking. Also if possible, we should document the skill change through the change in the parameters, ideally with headline scores but at least with mean MSE loss value.

File: config/default_config.yml (hunk 1 above)
Comment (SavvasMel) [reply config/default_config.yml:41]: So yes, this is unrelated and apologies that I forgot to mention it. The introduced changes in the network parameters will work well for both cases. I changed it since pretraining on autoencoder-decoder style with masking is the first step now anyway. But I can change it back of course.

**Comment by mlangguth89** [comment]
I think we should also change the default value of `token_size` in the ERA5 stream file.
With the used dataset, we just have 2-5 input items per healpix cell as documented in the [hedgedoc](https://gitlab.jsc.fz-juelich.de/hedgedoc/xuYwPs8qRNa5PRvoK_oTyg#) related to #126. So, to avoid excessive zero-padding, we should lower `token_size`. The hedgedoc in #126 used 16, but one could probably also set to 8.

**Comment by mlangguth89** [comment]
> > I think we should also change the default value of `token_size` in the ERA5 stream file. With the used dataset, we just have 2-5 input items per healpix cell as documented in the [hedgedoc](https://gitlab.jsc.fz-juelich.de/hedgedoc/xuYwPs8qRNa5PRvoK_oTyg#) related to #126. So, to avoid excessive zero-padding with the [current default](https://github.com/ecmwf/WeatherGenerator/blob/3bb9b3858739d8842a79f847f6296f0ad0cf4c74/config/streams/streams_anemoi/era5.yml#L19), we should lower `token_size`. The hedgedoc in #126 used 16, but one could probably also set to 8.
> 
> Did we do experiments that compare token size versus healpix level? The latter is likely the better option, but we should base the decision on numbers.

Not really. Kerem documented the statistics for input items per cell for the ERA5 O96 and N320 data in the hedgedoc. This shows that the default value of  `token_size: 32` is quite large for the ERA5 O96 dataset that is utilized in the default experiments. Even with Healpix level 4 grid, one has mostly less or equal than 16 input items per grid cell, and thus we decided to stick to `token_size: 16`. Yet, a systematic analysis of token size vs. Healipix level remains to be done, but for the standard experiments, the current default value does not make much sense. The input tensors to the embedding network are just inflated with many padded zeros (typically about 28 zeros per cell/channel).

**Comment by SavvasMel** [comment]
@clessig @mlangguth89 @tjhunter @kctezcan The introduced changes are motivated by the recent runs that I had based on issue https://github.com/ecmwf/WeatherGenerator/issues/126 (and the respective hedgedoc) and that were replicated by @kctezcan very recently, look for example the training loss plot below that was uploaded in the "ERA5 training optimization" channel:

image

This is related also to issue and comments https://github.com/ecmwf/WeatherGenerator/issues/242#issuecomment-2909927558 and https://github.com/ecmwf/WeatherGenerator/issues/242#issuecomment-2919824673 where we can see that the performance by just using the "default" parameters (reported by @kctezcan) is not as good as the performance that is reported later on (by me). In my case, the parameters that have been introduced in this PR have been used.

I should note that the above is not related to the the `offset` parameter. We can still leave this to be `1`.
